code/API_definitions/device-network-type-subscription.yaml

```diff
@@ -0,0 +1,695 @@
+          $ref: "#/components/schemas/Source"
+        type:
+          $ref: "#/components/schemas/NotificationEventType"
+        specversion:
+          type: string
+          description: Version of the specification to which this event conforms (must be 1.0 if it conforms to cloudevents 1.0.2 version)
+          example: "1.0"
+        datacontenttype:
+          type: string
+          description: 'media-type that describes the event payload encoding, must be "application/json" for CAMARA APIs'
+          example: application/json
+        data:
+          type: object
+          description: Event details payload described in each CAMARA API and referenced by its type
+        time:
+          $ref: "#/components/schemas/EventTime"
+      discriminator:
+        propertyName: "type"
+        mapping:
+          org.camaraproject.connected-network-type.v0.network-type-changed: "#/components/schemas/EventNetworkTypeChange"
+
+    Source:
+      type: string
+      format: uri-reference
+      minLength: 1
+      description: |
+        Identifies the context in which an event happened, as a non-empty `URI-reference` like:
+        - URI with a DNS authority:
+          * https://github.com/cloudevents
+          * mailto:cncf-wg-serverless@lists.cncf.io
+        - Universally-unique URN with a UUID:
+          * urn:uuid:6e8bc430-9c3a-11d9-9669-0800200c9a66
+        - Application-specific identifier:
+          * /cloudevents/spec/pull/123
+          * 1-555-123-4567
+      example: "https://notificationSendServer12.supertelco.com"
+
+    EventTime:
+      type: string
+      format: date-time
+      description: |
+        Timestamp of when the occurrence happened.
+        If the time of the occurrence cannot be determined then this attribute MAY be set to some other time (such as the current time) by the CloudEvents producer,
+        however all producers for the same source MUST be consistent in this respect. In other words,
+        either they all use the actual time of the occurrence or they all use the same algorithm to determine the value used.
+        It must follow [RFC 3339](https://datatracker.ietf.org/doc/html/rfc3339#section-5.6) and must have time zone.
+        Recommended format is yyyy-MM-dd'T'HH:mm:ss.SSSZ (i.e. which allows 2023-07-03T14:27:08.312+02:00 or 2023-07-03T12:27:08.312Z)
+      example: "2018-04-05T17:31:00Z"
+
+
+    EventNetworkTypeChange:
+      description: event structure for network type change
+      allOf:
+        - $ref: "#/components/schemas/CloudEvent"
+        - type: object
+          properties:
+            data:
+              $ref: "#/components/schemas/BasicDeviceEventData"
+
+    BasicDeviceEventData:
+      description: Event detail structure for basic device events
+      type: object
+      required:
+        - device
+      properties:
+        device:
+          $ref: "#/components/schemas/Device"
+        subscriptionId:
+          $ref: "#/components/schemas/SubscriptionId"

```

Comment:
We should also add the information about what the new connectedNetworkType is, right?
Otherwise the user just retrieves an event that the network type changed but no information about what was changed.

And making all of these properties mandatory.

```yaml
    EventNetworkTypeChange:
      description: event structure for network type change
      allOf:
        - $ref: "#/components/schemas/CloudEvent"
        - type: object
          properties:
            data:
              $ref: "#/components/schemas/EventNetworkTypeChangeData"

    EventNetworkTypeChangeData:
      description: Event detail structure for basic device events
      type: object
      required:
        - device
        - connectedNetworkType
        - subscriptionId
      properties:
        device:
          $ref: "#/components/schemas/Device"
        connectedNetworkType:
          $ref: "#/components/schemas/ConnectedNetworkType"
        subscriptionId:
          $ref: "#/components/schemas/SubscriptionId"
```